networkx/utils/tests/test_decorators.py

```diff
@@ -489,3 +489,29 @@ def nop(self):
         print(foo.__argmap__.assemble(foo.__wrapped__))
         argmap._lazy_compile(foo)
         print(foo._code)
+
+    def test_immediate_raise(self):
+        @not_implemented_for("directed")
+        def yield_nodes(G):
+            yield from G
+
+        G = nx.Graph([(1, 2)])
+        D = nx.DiGraph()
+
+        print([*yield_nodes(G)])
+        print(yield_nodes._code)
```

Comment:
I have a slight preference for swapping out the print function with something more test-y, e.g. in the first case maybe something like:

```
assert list(yield_nodes(G)) == [1, 2]
```

I don't fully grok what the second print tests, but if it's just the existence of the `_code` attr, maybe replace with an `assert hasattr`?

Reply:
Good catch; the prints were only meant to be temporary.